deploy/template.yaml

```diff
@@ -1220,6 +1271,46 @@ Resources:
                   - "dynamodb.amazonaws.com"
                   - "ecs-tasks.amazonaws.com"
 
+  S3KmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - kms:*
+            Resource: "*"
+          - Effect: Allow
+            Principal:
+              AWS: !FindInMap
+                - EnvironmentConfiguration
+                - !Ref AWS::AccountId
+                - spotAccountNumber
+            Action:
+              - "kms:Encrypt*"
```

Comment:
I think just `kms:Encrypt` and `kms:Decrypt` is sufficient.

Reply:
Probably needs `kms:GenerateDataKey` too.

Reply:
Actually, it turns out we don't need `kms:Encrypt` in this instance, see the [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingKMSEncryption.html).

`kms:GenerateDataKey` is required for encrypting the data to upload to the bucket using `s3:PutObject`, `kms:Decrypt` is required to download using `s3:GetObject`.

Reply:
Spot account perms are now decrypt, and generateDataKey